create gyroidos-guestos.bbclass for common things

jwsbck opened the pull request:
This PR makes four contributions:
1. Create common class gyroid-guestos.bbclass as base for all guestos recipes.
2. Migrate trustx-core to the new gyroid-guestos class.
3. Move deployment of device.conf from guestos recipes to trustx-cml.
4. Move all affected config files from gyroidos_build to yocto layers.

associated pull requests:
https://github.com/gyroidos/meta-trustx-intel/pull/27
https://github.com/gyroidos/gyroidos_build/pull/107

@quitschbo: How shall we handle other images (i.e. deb.bb, docker-convert.bb and ids.bb)? Shall we convert them to the new gyroid-guestos class as well?

quitschbo commented:
Looks good. And yes deb.bb and docker-convert.bb should be migrated, too. The ids.bb, we could drop completely.
Also the recipes to the java based trusted connector could be removed. I think we already dropped meta-java anyway, didn't we?

jwsbck commented:
Alright. I dropped the ids related recipes and removed meta-java. I think we did plan this a while ago but did not actually remove it in the end.

Regarding debos and docker-convertos: i had a look at the recipes and noticed that bot share exactly one LoC with the trustx-coreos recipe. Given that, i would probably leave them as they are for now. Perhaps the recipes should be reworked/cleaned up one day but i need more insight about how they actually work than i have at the moment.
